development-vm/replication/sync-aws-elasticsearch.sh

```diff
@@ -61,42 +67,46 @@ else
 fi
 
 if [ $# -gt 0 ]; then
-  filenames=""
-  for index_name in $@; do
-    filenames="$filenames $LOCAL_ARCHIVE_PATH/$index_name.zip"
-  done
-
-  for filename in $filenames; do
-    if [ ! -e $filename ]; then
-      error "File $filename not found: aborting."
-      exit 1
-    fi
-  done
+  possible_names=$@
 else
-  filenames=${LOCAL_ARCHIVE_PATH}/*.zip
+  possible_names=$(ls ${LOCAL_ARCHIVE_PATH}/indices)
 fi
 
-status "Restoring data into Elasticsearch"
-
-for f in $filenames
-do
-  if $DRY_RUN; then
-    status "$f (dry run)"
-  else
-    if curl $LOCAL_ES_HOST -o /dev/null 2> /dev/null; then
-      ok "Elasticsearch is running on ${LOCAL_ES_HOST}"
-    else
-      error "Elasticsearch is not running on ${LOCAL_ES_HOST}. Aborting..."
-      exit 1
-    fi
-
-    alias_name=$(basename $f .zip)
-    iso_date="$(date --iso-8601=seconds|cut --byte=-19|tr [:upper:] [:lower:])z"
-    real_name="$alias_name-$iso_date-00000000-0000-0000-0000-000000000000"
-
-    status "Importing $f to $alias_name (real name $real_name)"
-    bundle exec es_dump_restore restore_alias "$LOCAL_ES_HOST" "$alias_name" "$real_name" "$f" '{"settings":{"index":{"number_of_replicas":"0","number_of_shards":"1"}}}' 250
+index_names=""
+first_name=1
+for index_name in $possible_names; do
+  if [ $first_name -lt 1 ]; then
+    index_names="$index_names,"
   fi
+  first_name=0
+  index_names="$index_names$(ls ${LOCAL_ARCHIVE_PATH}/indices | grep ${index_name})"
 done
 
+if $DRY_RUN; then
+  status "Restoring data into Elasticsearch for $index_names (dry run)"
+else
+  status "Restoring data into Elasticsearch for $index_names"
+
+  # setup the snapshot details on the server
+  curl localhost:9200/_snapshot/snapshots?wait_for_completion=true -X PUT -d "{
+    \"type\": \"fs\",
+    \"settings\": {
+      \"compress\": true,
+      \"location\": \"/var/govuk/govuk-puppet/development-vm/replication/${LOCAL_ARCHIVE_PATH}\"
+    }
+  }"
+
+  # restore the snapshot
+  curl localhost:9200/_snapshot/snapshots/snapshot_2018-02-07/_restore?wait_for_completion=true -X POST -d "{
+    \"indices\": \"${index_names}\",
+    \"index_settings\": {
+      \"index.number_of_replicas\": 0
```

Comment:
Is it OK to hard-code this, or can we get it from a puppet setting somewhere?

Reply:
I think so.